packages/flutter_tools/lib/src/doctor_validator.dart

```diff
@@ -162,47 +162,31 @@ class ValidationResult {
   final List<ValidationMessage> messages;
 
   String get leadingBox {
-    switch (type) {
-      case ValidationType.crash:
-        return '[☠]';
-      case ValidationType.missing:
-        return '[✗]';
-      case ValidationType.success:
-        return '[✓]';
-      case ValidationType.notAvailable:
-      case ValidationType.partial:
-        return '[!]';
-    }
+    return switch (type) {
+      ValidationType.crash   => '[☠]',
+      ValidationType.missing => '[✗]',
+      ValidationType.success => '[✓]',
+      ValidationType.notAvailable || ValidationType.partial => '[!]',
+    };
   }
 
   String get coloredLeadingBox {
-    switch (type) {
-      case ValidationType.crash:
-        return globals.terminal.color(leadingBox, TerminalColor.red);
-      case ValidationType.missing:
-        return globals.terminal.color(leadingBox, TerminalColor.red);
-      case ValidationType.success:
-        return globals.terminal.color(leadingBox, TerminalColor.green);
-      case ValidationType.notAvailable:
-      case ValidationType.partial:
-        return globals.terminal.color(leadingBox, TerminalColor.yellow);
-    }
+    return globals.terminal.color(leadingBox, switch (type) {
+      ValidationType.success => TerminalColor.green,
+      ValidationType.crash || ValidationType.missing => TerminalColor.red,
+      ValidationType.notAvailable || ValidationType.partial => TerminalColor.yellow,
+    });
   }
 
   /// The string representation of the type.
   String get typeStr {
-    switch (type) {
-      case ValidationType.crash:
-        return 'crash';
-      case ValidationType.missing:
-        return 'missing';
-      case ValidationType.success:
-        return 'installed';
-      case ValidationType.notAvailable:
-        return 'notAvailable';
-      case ValidationType.partial:
-        return 'partial';
-    }
+    return switch (type) {
```

Comment:
could just have the `success` -> `installed` bit and then a default to using `name`

Reply:
Won't the `default` make it so that the lint about not using all of the enum values doesn't fire if a new enum value is added (which could have a type name that doesn't match the enum name). Without the default, the lint will require considering this switch when one is added.

Reply:
Yep, I like the intention behind this suggestion, but it doesn't conform to the style guidelines.

But I still think it's possible to get rid of the switch expression altogether:

```dart
enum ValidationType {
  crash,
  missing,
  partial,
  notAvailable,
  success('installed');

  const ValidationType([this.label]);
  final String? label;

  @override
  String toString() => label ?? name;
}
```